scenes/game/ai_player.gd

```diff
@@ -450,29 +450,37 @@ func get_ex_option_in_hand(game_logic : LocalGame, me : LocalGame.Player, card_i
 			return card.id
 	return -1
 
-func get_strike_actions(game_logic : LocalGame, me : LocalGame.Player, _opponent : LocalGame.Player):
+func get_strike_actions(game_logic : LocalGame, me : LocalGame.Player, _opponent : LocalGame.Player, from_gauge : bool = false):
 	var possible_actions = []
-	# Always allow wild swing.
-	possible_actions.append(StrikeAction.new(-1, -1, true))
+	# Always allow wild swing if allowed.
+	if not from_gauge:
+		possible_actions.append(StrikeAction.new(-1, -1, true))
 
 	# Ignore cards that you can't pay for.
 	# Wait to pay for cards until later (you get to see what they flip).
 	# Add wild swing.
 	var added_ex_options = [-1]
-	for card in me.hand:
-		if card.definition['gauge_cost'] > me.gauge.size():
-			# Skip cards we can't pay for.
-			continue
-		var ex_card_id = get_ex_option_in_hand(game_logic, me, card.id)
-		if ex_card_id not in added_ex_options and card.id not in added_ex_options:
-			# If we can play EX, add that as an option.
-			# Don't consider ex again for these cards.
-			added_ex_options.append(ex_card_id)
-			added_ex_options.append(card.id)
-			possible_actions.append(StrikeAction.new(card.id, ex_card_id, false))
+	if from_gauge:
+		for card in me.gauge:
+			if card.definition['gauge_cost'] > me.gauge.size() - 1:
+				# Skip cards we can't pay for.
+				continue
```

Comment:
I think it's short circuiting to wild swing properly instead of asking for an ai action if the gauge is empty; I can add this as a failsafe just in case though

Reply:
But you could have like, just a card in your gauge you can't pay for right?
If you're not accounting for that the AI will still try to do it.

Reply:
Oh that's true; I'll remove the check for gauge cost since that's technically a legal option, but hopefully the other change to the wild swing logic on this will help cover other weird cases